### PR TITLE
Bugfixes for connectors

### PR DESCRIPTION
**Pull request by josvandervelde**
1. Updating example connector resources, to include all fields (mainly educational resource json)
2. Bugfixes for HuggingFace, Openml and Zenodo. They all run fine now.
3. To make that possible, I changed all `NamedRelations` to be always lowercase. E.g. keyword. This means that keyword "AI" is changed into "ai". This fixes some weird errors --> using `mysql` we got an integrity error if you first add "AI" and then "ai", although "SELECT * FROM keyword WHERE name = "AI"` doesn't retrieve the keyword with name "ai". 
4. Some quality of life improvements: added profiles to the docker compose (making it easy to not run the connectors), and throwing existing state of connectors away if the database is empty

In another PR I'll fix the "Permission denied" errors that you get when running the code for the first time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,13 @@ mysql> SHOW DATABASES;
 
 Now, you can visit the server from your browser at `localhost:8000/docs`.
 
+#### Using connectors
+You can specify different connectors using
+
+```bash
+docker compose --profile examples --profile huggingface-datasets --profile openml-datasets up -d
+docker compose --profile examples --profile huggingface-datasets --profile openml-datasets down
```
Comment (Taniya-Das) [inline README.md:101]: Not sure if necessary -  mention use of --profile for docker compose down as well, else some container will still be up after docker compose down.

File: src/connectors/synchronization.py [hunk 2]
```diff
@@ -183,9 +185,7 @@ def main():
                 logging.info(f"Saving state after handling {i}th result: {json.dumps(state)}")
                 with open(state_path, "w") as f:
                     json.dump(state, f, indent=4)
```
Comment (Taniya-Das) [inline src/connectors/synchronization.py:187]: Why session.commit() is deleted from here, though new data is getting written?

**Review by Taniya-Das — APPROVED** [review]
Looks good

File: README.md (hunk 1 above)
Comment (josvandervelde) [reply README.md:107]: Thanks, good point! I fixed it in https://github.com/aiondemand/AIOD-rest-api/pull/191/commits/bfb0d5e21c521a964a33649cd77244a6fb78c5ce

**Pull request merged**